go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -204,95 +204,105 @@ func (rs *rowStreamer) buildSelect() (string, error) {
 }
 
 func (rs *rowStreamer) streamQuery(conn *snapshotConn, send func(*binlogdatapb.VStreamRowsResponse) error) error {
-	log.Infof("Streaming query: %v\n", rs.sendQuery)
-	gtid, err := conn.streamWithSnapshot(rs.ctx, rs.plan.Table.Name, rs.sendQuery)
-	if err != nil {
-		return err
-	}
-
-	// first call the callback with the fields
-	flds, err := conn.Fields()
-	if err != nil {
-		return err
-	}
-	pkfields := make([]*querypb.Field, len(rs.pkColumns))
-	for i, pk := range rs.pkColumns {
-		pkfields[i] = &querypb.Field{
-			Name: flds[pk].Name,
-			Type: flds[pk].Type,
-		}
-	}
-
-	err = send(&binlogdatapb.VStreamRowsResponse{
-		Fields:   rs.plan.fields(),
-		Pkfields: pkfields,
-		Gtid:     gtid,
-	})
-	if err != nil {
-		return fmt.Errorf("stream send error: %v", err)
-	}
 
-	response := &binlogdatapb.VStreamRowsResponse{}
-	lastpk := make([]sqltypes.Value, len(rs.pkColumns))
-	byteCount := 0
+	offset := 0
+	limit := 10000000
+	delta := 10000001
 	for {
-		//log.Infof("StreamResponse for loop iteration starts")
-		select {
-		case <-rs.ctx.Done():
-			log.Infof("Stream ended because of ctx.Done")
-			return fmt.Errorf("stream ended: %v", rs.ctx.Err())
-		default:
+		rs.sendQuery = fmt.Sprintf("%s LIMIT %d OFFSET %d", rs.sendQuery, limit, offset)
+		log.Infof("Streaming query: %v\n", rs.sendQuery)
+		gtid, err := conn.streamWithSnapshot(rs.ctx, rs.plan.Table.Name, rs.sendQuery)
```

Comment:
This needs to get refactored to avoid setting the isolation level and starting a transaction multiple times on the same connection: I think it ends up being _fine_ since mysql will tell you you can't do this but we end up locking the table multiple times for no reason so it'd be great to untangle that.

Reply:
another note that i didn't even realize the first time around: after the first iteration of this loop, you'll be tacking on limit offset suffixes to a statement that already has them:

```
# iteration 1
sendQuery = "select * from table"
sendQuery += sendQuery + "limit $limit offset $offset"

# iteration two
sendQuery = "select * from table limit $limit offset $offset"
sendQuery += sendQuery + "limit $limit offset $offset"
```